spring-boot-project/spring-boot-tools/spring-boot-buildpack-platform/src/main/java/org/springframework/boot/buildpack/platform/docker/configuration/DockerConfiguration.java

```diff
@@ -63,16 +63,17 @@ public DockerRegistryAuthentication getPublishRegistryAuthentication() {
 		return this.publishAuthentication;
 	}
 
-	public DockerConfiguration withHost(String address, boolean secure, String certificatePath) {
+	public DockerConfiguration withHost(String address, boolean secure, String certificatePath, Integer socketTimeout) {
 		Assert.notNull(address, "Address must not be null");
-		return new DockerConfiguration(DockerHostConfiguration.forAddress(address, secure, certificatePath),
+		return new DockerConfiguration(
+				DockerHostConfiguration.forAddress(address, secure, certificatePath, socketTimeout),
 				this.builderAuthentication, this.publishAuthentication, this.bindHostToBuilder);
 	}
 
-	public DockerConfiguration withContext(String context) {
+	public DockerConfiguration withContext(String context, Integer socketTimeout) {
```

Comment:
This is a breaking API change. A separate method for the timeout would be better.